### PR TITLE
Fix issue with ForeignKey creation

### PR DESCRIPTION
**Pull request by dreab8**
…g#forEachJdbcValue method

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java [hunk 1]
```diff
@@ -1048,6 +1052,30 @@ else if ( fkTarget instanceof EmbeddableValuedModelPart ) {
 		}
 	}
 
+	private static void setReferencedEmbeddableForeignKeyDescriptor(
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java:1055]: IMO this method should be removed i.e. revert to using `buildEmbeddableForeignKeyDescriptor`. Either this method does the same as `buildEmbeddableForeignKeyDescriptor`, except for calling `setForeignKeyDescriptor`, or `buildEmbeddableForeignKeyDescriptor` is using the wrong selection mapping for the target columns and it should be fixed. Or am I missing something?

**Review by beikov — CHANGES_REQUESTED** [review]

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java:1055]: it means to make tobuildEmbeddableForeignKeyDescriptor more complex, decide about the target of referenced columns and selections.

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java:1055]: in this case we can just inline it

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java [hunk 2]
```diff
@@ -653,6 +653,12 @@ public int forEachDisassembledJdbcValue(Object value, Clause clause, int offset,
 
 	@Override
 	public int forEachJdbcValue(Object value, Clause clause, int offset, JdbcValuesConsumer consumer, SharedSessionContractImplementor session) {
-		return foreignKeyDescriptor.forEachJdbcValue( value, clause, offset, consumer, session );
+		return foreignKeyDescriptor.forEachJdbcValue(
+				entityMappingType.getEntityPersister().getIdentifier( value, session ),
```
[the remark this answers is not in the document]
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java:657]: it does not work with `getPropertyAccess().getGetter().get(value)`

**Comment by beikov** [comment]
Thanks, but as discussed, I'm closing this in favor of https://github.com/hibernate/hibernate-orm/pull/3735

**Pull request closed without merging**